### PR TITLE
[MSFT] New dialect for Microsoft internal design support

### PR DESCRIPTION
**Pull request by teqdruid**
Umbrella dialect for everything needed to support Microsoft development but not thoroughly discussed. Most (if not everything) in this dialect is a candidate for generalization and re-homing.

### CODE REVIEW

File: include/circt/Dialect/MSFT/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+##===- CMakeLists.txt - MSFT dialect build definitions --------*- cmake -*-===//
+##
+## Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+## See https://llvm.org/LICENSE.txt for license information.
+## SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+##
+##===----------------------------------------------------------------------===//
+##
+##
+##===----------------------------------------------------------------------===//
```
Comment (lattner) [inline include/circt/Dialect/MSFT/CMakeLists.txt:10]: nit: Plz drop the bottom block of the head comment if it doesn't contain anything.

File: include/circt/Dialect/MSFT/MSFTDialect.h [hunk 2]
```diff
@@ -0,0 +1,36 @@
+//===- MSFTDialect.h - PD dialect declaration -------------------*- C++ -*-===//
```
Comment (lattner) [inline include/circt/Dialect/MSFT/MSFTDialect.h:1]: s/PD/something else

File: lib/Dialect/MSFT/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,30 @@
+##===- CMakeLists.txt - build definitions for MSFT ------------*- cmake -*-===//
+##
+## Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+## See https://llvm.org/LICENSE.txt for license information.
+## SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+##
+##===----------------------------------------------------------------------===//
+##
+##
+##===----------------------------------------------------------------------===//
```
Comment (lattner) [inline lib/Dialect/MSFT/CMakeLists.txt:10]: ditto

**Review by lattner — APPROVED** [review]
Ok.  Please also add something to circt/codeowners/

File: include/circt/Dialect/MSFT/MSFTDialect.h (hunk 2 above)
Comment (teqdruid) [reply include/circt/Dialect/MSFT/MSFTDialect.h:1]: Damn! Missed that.

**Comment by teqdruid** [comment]
@lattner Thanks for the fine tooth comb over!

**Comment by lattner** [comment]
happy to help

**Pull request merged**